### PR TITLE
added Unit test for the table-of-contents block and body.html as mock

### PR DESCRIPTION
**Pull request by skholkhojaev**
**Fixes following issues|Adds following features:**
- Issue/Feature 1

**Resolves:** [MWPW-NUMBER](https://jira.corp.adobe.com/browse/MWPW-NUMBER)

**Steps to test the before vs. after and expectations:**
- Steps for feature 1

**Pages to check for regression and performance:**
- https://MWPW-159132-test-table-of-contents--express--adobecom.hlx.page/docs/library/kitchen-sink/table-of-contents?martech=off

### CODE REVIEW

File: test/unit/blocks/table-of-contents/table-of-contents.test.js [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+
+  it('should not generate toc-entry elements for headings above the configured level', () => {
+    const block = document.querySelector('.table-of-contents');
+    const config = { levels: '2' };
+    block.dataset.config = JSON.stringify(config);
+
+    decorate(block, 'table-of-contents', document);
+
+    const tocEntries = document.querySelectorAll('.toc-entry');
+    tocEntries.forEach(($entry) => {
```
Comment (vhargrave) [inline test/unit/blocks/table-of-contents/table-of-contents.test.js:28]: why use the $ for your variable names? What value is it bringing?

File: test/unit/blocks/table-of-contents/table-of-contents.test.js [hunk 2]
```diff
@@ -1,4 +1,33 @@
 /* eslint-env mocha */
 /* eslint-disable no-unused-vars */
 
+import { readFile } from '@web/test-runner-commands';
+import { expect } from '@esm-bundle/chai';
+
 const { default: decorate } = await import('../../../../express/blocks/table-of-contents/table-of-contents.js');
+
+describe('table-of-contents', () => {
+  before(() => {
```
Comment (vhargrave) [inline test/unit/blocks/table-of-contents/table-of-contents.test.js:10]: why are you doing this?

File: test/unit/blocks/table-of-contents/table-of-contents.test.js [hunk 3]
```diff
@@ -1,4 +1,33 @@
...
+    window.isTestEnv = true;
+  });
+
+  beforeEach(async () => {
+    document.body.innerHTML = await readFile({ path: './mocks/body.html' });
+    const block = document.querySelector('.table-of-contents');
+    block.classList.add('table-of-contents');
+  });
+
+  it('should not generate toc-entry elements for headings above the configured level', () => {
```
Comment (vhargrave) [inline test/unit/blocks/table-of-contents/table-of-contents.test.js:20]: This test name is a little wrong. 
The toc block cannot be passed any type of config, which will make h1 tags appear as title elements. 
I'd recommend renaming it to 
'generates toc-entry elements for h2 headings'

File: test/unit/blocks/table-of-contents/table-of-contents.test.js [hunk 4]
```diff
@@ -1,4 +1,33 @@
...
+    const block = document.querySelector('.table-of-contents');
+    const config = { levels: '2' };
```
Comment (vhargrave) [inline test/unit/blocks/table-of-contents/table-of-contents.test.js:22]: I'd love to see a test, where you're also testing that h2 & h3 headers get rendered successfully.

**Review by vhargrave — COMMENTED** [review]
Very nice job for your first test 👏 
I've been very nit-picky just to help you learn, but for other people I'd have just approved this PR and we'd already be moving on. Good job!

File: test/unit/blocks/table-of-contents/table-of-contents.test.js (hunk 3 above)
Comment (skholkhojaev) [reply test/unit/blocks/table-of-contents/table-of-contents.test.js:20]: Will change that

File: test/unit/blocks/table-of-contents/table-of-contents.test.js (hunk 1 above)
Comment (skholkhojaev) [reply test/unit/blocks/table-of-contents/table-of-contents.test.js:28]: i thought using $ would indicate that the variable contians a JQuery Dom element but now that i researched I realised that its not necessery and has no functional purpose

File: test/unit/blocks/table-of-contents/table-of-contents.test.js (hunk 2 above)
Comment (skholkhojaev) [reply test/unit/blocks/table-of-contents/table-of-contents.test.js:10]: i took the inspiration from other writted Test by other developers, but it mainly serves to prepare the environment before any Test runs

Comment (JingleH) [reply test/unit/blocks/table-of-contents/table-of-contents.test.js:10]: i think it's a legacy practice we used to have, which allows our application code to behave differently based on the environment. I have not been rigorous in gating people away from doing this, but it's unneeded

File: test/unit/blocks/table-of-contents/table-of-contents.test.js (hunk 1 above)
Comment (vhargrave) [reply test/unit/blocks/table-of-contents/table-of-contents.test.js]: yes, and with frontend code you're also sending unnecessary bytes to our clients, which takes bandwidth :)

**Review by vhargrave — APPROVED** [review]
nice!

**Pull request merged**